src/fairchem/core/models/finetune_hydra.py

```diff
@@ -0,0 +1,187 @@
+    STARTING_MODEL = "starting_model"
+    MODE = "mode"
+    HEADS = "heads"
+
+    def __init__(self, config: dict):
+        self.config = config
+        self._mode = FineTuneMode[self.config[FTConfig.MODE]]
+        assert (
+            (FTConfig.STARTING_CHECKPOINT in self.config)
+            or (FTConfig.STARTING_MODEL in self.config)
+        ), "Either a starting checkpoint or a starting model must be provided!"
+        assert FTConfig.MODE in self.config
+        if self._mode == FineTuneMode.RETAIN_BACKBONE_ONLY:
+            # in this mode, we keep the backbone but attach new output heads specified in head config
+            assert (
+                FTConfig.HEADS in self.config
+            ), "heads cannot be empty when using RETAIN_BACKBONE_ONLY mode!"
+
+    def load_model(self) -> nn.Module:
+        # if provided a hydra config to start, build from the starting hydra model
+        # this assumes the weights are loaded from the state_dict in the checkpoint.pt file instead
+        # so no actual weights are loaded here
+        if FTConfig.STARTING_MODEL in self.config:
+            # register model from hydra_config
+            config_copy = copy.deepcopy(self.config[FTConfig.STARTING_MODEL])
+            name = config_copy.pop("name")
+            hydra_model = registry.get_model_class(name)(**config_copy)
+            assert isinstance(hydra_model, HydraInterface)
+        # if provided a checkpoint to start then load the model and weights from the given checkpoint
+        # this happens used in the beginning of a finetuning run
+        elif FTConfig.STARTING_CHECKPOINT in self.config:
+            hydra_model: HydraInterface = load_hydra_model(
+                self.config[FTConfig.STARTING_CHECKPOINT]
+            )
+
+        num_params = sum(p.numel() for p in hydra_model.parameters())
+        logging.info(f"Loaded Original hydra model with {num_params} params")
+        return hydra_model
+
+    def get_standalone_config(self) -> dict:
+        # replace a config with a checkpoint with one that has the model config only
+        # this is required for standalone prediction (so we don't need to ship the original checkpoint),
+        # multi-round finetuning, and better robustness
+        standalone_config = {
+            "name": FTHYDRA_NAME,
+            FTConfig.FT_CONFIG_NAME: self.config,
+        }
+        if FTConfig.STARTING_CHECKPOINT in self.config:
+            # modify the config to store the original model config inside model attrs
+            # so we dont need the checkpoint again when loading from checkpoint
+            new_config = copy.deepcopy(self.config)
+            new_config[FTConfig.STARTING_MODEL] = (
+                get_model_config_from_checkpoint(
+                    self.config[FTConfig.STARTING_CHECKPOINT]
+                )
+            )
+            standalone_config[FTConfig.FT_CONFIG_NAME] = new_config
+            return standalone_config
```

Comment:
super nit; in either case return standalone_config , can that just be moved to outside the if?